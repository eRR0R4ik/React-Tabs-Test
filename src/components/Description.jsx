import React from 'react';
import PropTypes from 'prop-types';

const Description = ({
  picture,
  name,
  description
}) => {
  return (
    <div className="phases-img">
      <img
        src={picture}
        style={{ float: 'left', width: '250px', padding: '0 25px 0 0' }}
      />
      <h4>{name}</h4>
      <div
        dangerouslySetInnerHTML={{ __html: description }}
      />
    </div>
  );
}

Description.propTypes = {
  picture: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default Description;
