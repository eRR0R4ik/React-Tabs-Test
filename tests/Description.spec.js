import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Description from '../src/components/Description.jsx';

describe('Component: Description', () => {
  const minProps = {
    picture: 'mockPics',
    name: 'mockName',
    description: 'mockDescription'
  }

  it('should render <Description /> components', () => {
    expect(shallow(<Description {...minProps} />).length).toEqual(1);
  });

  it('should render correctly', () => {
    const component = shallow(
      <Description {...minProps} />
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
