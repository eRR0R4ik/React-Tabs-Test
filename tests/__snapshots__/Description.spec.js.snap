// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Description should render correctly 1`] = `
<div
  className="phases-img"
>
  <img
    src="mockPics"
    style={
      Object {
        "float": "left",
        "padding": "0 25px 0 0",
        "width": "250px",
      }
    }
  />
  <h4>
    mockName
  </h4>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "mockDescription",
      }
    }
  />
</div>
`;
